RawCMS.Library/Lambdas/JsDispatcher.cs

```diff
@@ -28,7 +29,7 @@ public JsDispatcher(EntityService entityService)
             this.entityService = entityService;
         }
 
-        public override void Execute(string collection, ref JObject item, ref Dictionary<string, object> dataContext)
+        public override void ExecuteInternal(string collection, ref JObject item, ref Dictionary<string, object> dataContext)
         {
```

Comment:
This method take two more parameters: the operation and stage from Execute method (see event classes). Well, basing on the event and the stage it find the right js code to execute.

This piece of code execute anytime the presave script with engine.Execute(settings.PresaveScript);, but

If operation is write and stage is pre then you will need to use PresaveScritpt. 
If operation is write and stage is post then you will need to use PostsaveScritpt, and so on, as in task description.

Moreover, we need to have multiple event in settings. Events.XXX and model is still the old version.